src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Utf8JsonReader.Comments.Tests.cs

```diff
@@ -0,0 +1,138 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace System.Text.Json.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public partial class Utf8JsonReaderCommentsTests
+    {
+        [Params(JsonCommentHandling.Skip, JsonCommentHandling.Allow)]
+        public JsonCommentHandling CommentHandling;
+
+        // 0 => single segment
+        [Params(0, 100)]
+        public int SegmentSize;
+
+        private bool MultiSegment => SegmentSize != 0;
```

Comment:
I think this bool is what we should have as the Params (true for single segment, false for multisegment), rather than SegmentSize.

1) SegmentSize is confusing since 0 is supposed to mean single segment (but the person reviewing the perf results wouldn't know that).
2) SegmentSize of 100 is inaccurate since based on the payload, we won't have 100 segments.
3) It's only used in GobalSetup, whereas the bool is used in the actual test.

```C#
[Params(true, false)]
public bool SingleSegment;

[GlobalSetup]
public void Setup()
{
   _jsonPayload = Encoding.UTF8.GetBytes(s_testCases[TestCase]);
   _jsonPayloadSequence = GetSequence(_jsonPayload, 100);
}
```

That way, the perf results are more meaningful, imo (and doesn't require additional mapping to various concepts which require looking at perf test implementation).

Reply:
For me interesting optimization point is also SegementSize=1, with current state I can quickly add that case

Reply:
If we stick to the test we have atm, having `SegmentSize` as a column in the test result doesn't make sense. Having SingleSegment/MultiSegment as a column makes more sense.

Alternatively, we add two separate tests - one for single segment, and one for multi-segment.

> For me interesting optimization point is also SegementSize=1

In reality, we should only optimize for segment/buffer sizes of 4k/8k, and not 1-100 byte segments. I don't see much value in measuring the perf, and optimizing for, any unreasonable segment sizes.